packages/nimble-components/src/table/specs/table-column-width-hld.md

```diff
@@ -15,27 +15,45 @@ We need to provide users the means for changing the widths of individual columns
--   If a user drags a divider between two columns to the right, then the column on the left will grow larger, and the column on the right will grow smaller by the same pixel amount. Sub-behaviors to this are:
-    -   If a shrinking column has reached its minimum pixel size or is not resizable, then the next column in the direction of the sizing action will be affected up to the final column in a given direction.
-    -   A sizing action to the left will ultimately stop having an effect when the left-most column reaches its minimum size.
-    -   A sizing action to the right that would ultimately result in the final right column reaching its minimum size (all columns still within current table width) would then begin to push columns out of the table width resulting in a horizontal scrollbar.
+-   If a user drags a divider between two columns to the right, then the column on the left will grow larger, and the column on the right will grow smaller by the same pixel amount. Sub-behaviors to this are: - If a shrinking column has reached its minimum pixel size or is not resizable, then the next column in the direction of the sizing action will be affected up to the final column in a given direction. - A sizing action to the left will ultimately stop having an effect when the left-most column reaches its minimum size. - A sizing action to the right that would ultimately result in the final right column reaching its minimum size (all columns still within current table width) would _conditionally_ (see behavior descriptions below) begin to push columns out of the table viewport width resulting in a horizontal scrollbar on the table. - Within a single mouse interaction (i.e. drag-sizing without releasing mouse), if a cascade results in a column not adjacent to the divider being sized, then moving the mouse back in the opposite direction will "revert" the size made to the non-adjacent column.
+    ![Column resizing](spec-images/tableColumnResize.gif)
+
 -   Columns can be configured to not allow a user to interactively size them
     -   The implicit behavior present based on the behaviors described above, is that in a sizing action that cascades to a column configured to not be resized is that the column won't be resized towards a minimum size, and the cascade will effectively "skip" this column.
+-   A client will be able to configure the table such that dragging a column will never result in a change in the table's viewport width. Instead, once all columns have been reduced to their minimum width to the right of the divider, further mouse moves to the right will result in no action. This does _not_ mean that a table configured in such a way will _never_ have a scrollbar (e.g. a table could still have enough columns in its view to demand a horizontal scrollbar due to their combined minimum sizes).
```

Comment:
Is there a recommendation or idea of why a client would use one mode rather than another? I'm all for having the API be written in a way where we can add new modes in the future without breaking the existing API, but I'm not convinced we should add modes now that we don't have (even theoretical) use cases for.

Reply:
Agreed. I've removed this.